stages/org.osbuild.grub.iso

```diff
@@ -0,0 +1,160 @@
+  "properties": {
+    "product": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name", "version"],
+      "properties": {
+        "name": {"type": "string"},
+        "version": {"type": "string"}
+       }
+    },
+    "kernel": {
+      "type": "string"
+    },
+    "isolabel": {
+      "type": "string"
+    },
+    "efi": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["architectures", "vendor"],
+      "properties": {
+        "architectures": {
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "vendor": {
+            "type": "string"
+        }
+      }
+    },
+    "kernel_opts": {
+      "description": "Additional kernel boot options",
+      "type": "string"
+    },
+    "templates": {
+      "type": "string",
+      "default": "/usr/share/lorax/templates.d/99-generic"
+    }
+  }
+}
+"""
+
+def replace(target, patterns):
+    finder = [(re.compile(p), s) for p, s in patterns]
+    newfile = target + ".replace"
+
+    with open(target, "r") as i, open(newfile, "w") as o:
+        for line in i:
+            for p, s in finder:
+                line = p.sub(s, line)
+            o.write(line)
+    os.rename(newfile, target)
+
+def main(root, options):
+    name = options["product"]["name"]
+    version = options["product"]["version"]
+    isolabel = options["isolabel"]
+    templates = options["templates"]
+    efi = options.get("efi")
+    kopts = options.get("kernel_opts")
+
+    # input directories
+    configdir = os.path.join(templates, "config_files", "x86")
+
+    # output directories
+    imgdir = os.path.join(root, "images")
+    pxedir = os.path.join(imgdir, "pxeboot")
+
+    os.makedirs(imgdir)
+
+    cmdline = f""
+    if kopts:
+        cmdline += " " + kopts
+
+    info = {
+        "version": version,
+        "name": name,
+        "isolabel": isolabel,
+        "configdir": configdir,
+        "kerneldir": pxedir,
+        "imgdir": imgdir,
+        "cmdline": cmdline
+    }
+
+    arches = efi["architectures"]
+    vendor = efi["vendor"]
+
+    efidir = os.path.join(root, "EFI", "BOOT")
+    os.makedirs(efidir)
+
+    #arch related data
+    for arch in arches:
+        arch = arch.lower()
+        targets = [
+            (f"shim{arch}.efi", f"BOOT{arch}.EFI".upper()),
+            (f"mm{arch}.efi", f"mm{arch}.efi"),
+            (f"gcd{arch}.efi", f"grub{arch}.efi")
+        ]
+
+        for src, dst in targets:
+            shutil.copy2(os.path.join("/boot/efi/EFI/", vendor, src),
+                         os.path.join(efidir, dst))
+
+    # the font
+    fontdir = os.path.join(efidir, "fonts")
+    os.makedirs(fontdir, exist_ok=True)
+    shutil.copy2("/usr/share/grub/unicode.pf2", fontdir)
+
+    # the config
+    configdir = info["configdir"]
+    version = info["version"]
+    name = info["name"]
+    isolabel = info["isolabel"]
+    cmdline = info["cmdline"]
```

Comment:
We don't need to put everything in `info` and then pull it out of it again. Can just use the variables directly.